theano/gof/opt.py

```diff
@@ -2662,27 +2673,39 @@ def check_stack_trace(f_or_fgraph, ops_to_check='last', bug_print='raise'):
         else:
             raise ValueError('The string ops_to_check is not recognised')
 
-    elif hasattr(ops_to_check, '__call__'):  # if ops_to_check is a function
+    # if ops_to_check is a list/tuple of ops
+    elif isinstance(ops_to_check, (tuple, list)):
         apply_nodes_to_check = [node for node in fgraph.apply_nodes
-                                if ops_to_check(node)]
+                                if node.op in ops_to_check or
+                                isinstance(node.op, ops_to_check) or
+                                (hasattr(node.op, 'scalar_op') and
+                                 isinstance(node.op.scalar_op, ops_to_check))]
```

Comment:
Apparently, you can only have classes or types in the second argument of `isinstance`, so I think you are going need to either separate instances from Ops first, or iterate over the content of `ops_to_check` explicitly for each node.